aip/0233.md

```diff
@@ -0,0 +1,82 @@
+---
+aip:
+  id: 233
+  state: reviewing
+  created: 2019-06-18
+permalink: /233
+redirect_from:
+  - /0233
+---
+
+# Batch Create
+
+Some APIs need to allow users to create multiple resources in a single
+transaction. A batch create method provides this functionality.
+
+## Guidance
+
+APIs **may** support Batch Create using the following pattern:
+
+```proto
+rpc BatchCreateBooks(BatchCreateBooksRequest) returns (BatchCreateBooksResponse) {
+  option (google.api.http) = {
+    post: "/v1/shelves/-/books:batchCreate"
+    body: "*"
+  };
+}
+```
+
+- The RPC's name **must** begin with `BatchCreate`. The remainder of
+  the RPC name **should** be the plural form of the resource being created.
+- The request and response messages **must** match the RPC name, with `-Request`
+  and `-Response` suffixes.
+  - However, in the event that the request may take a significant amount of
```

Comment:
Is there a definition for "significant"? Or is it more descriptive to say "may exceed your SLA".

Reply:
I was matching verbiage from elsewhere - @lukesneeringer do we want to be more specific about this everywhere it's mentioned?

Reply:
Inclined against for the moment; SLA is a Google term (although many other companies have them also). Also, SLAs are likely to be significantly higher -- the rule of thumb for an LRO is 10 seconds, but SLAs are often 45+ seconds.

Reply:
Note: Resolving this conversation for now, but @alin04, if you are unconvinced, please bring it back up on the approval CL. :-)